service/tbc/websocket.go

```diff
@@ -47,6 +50,69 @@ func (ws *tbcWs) handlePingRequest(ctx context.Context, payload any, id string)
 	return nil
 }
 
+func (ws *tbcWs) handleBtcBlockMetadataByNumRequest(ctx context.Context, payload any, id string, db tbcd.Database) error {
+	log.Tracef("handleBtcBlockMetadataByNumRequest: %v", ws.addr)
+	defer log.Tracef("handleBtcBlockMetadataByNumRequest exit: %v", ws.addr)
+
+	writeHandleBtcBlockMetadataByNumResponse := func(res tbcapi.BtcBlockMetadataByNumResponse) error {
+		if err := tbcapi.Write(ctx, ws.conn, id, res); err != nil {
+			return fmt.Errorf("handleBtcBlockMetadataByNumRequest write: %v %v",
+				ws.addr, err)
+		}
+
+		return nil
+	}
+
+	p, ok := payload.(*tbcapi.BtcBlockMetadataByNumRequest)
+	if !ok {
+		return fmt.Errorf("handleBtcBlockMetadataByNumRequest invalid payload type: %T", payload)
+	}
+
+	bh, err := db.BlockHeadersByHeight(ctx, uint64(p.Height))
```

Comment:
@max-sanchez @marcopeereboom what do we do if we have more than 1 block header at this height? 🤔

Reply:
resolved with this for now https://bloqexternal.slack.com/archives/C06JTUUS9RA/p1711470734212029